main.go

```diff
@@ -71,6 +71,74 @@ func failf(format string, v ...interface{}) {
 	os.Exit(1)
 }
 
+// BuildGradleVersionUpdater updates versionName and versionCode in the given build.gradle file.
+type BuildGradleVersionUpdater struct {
+	buildGradleReader io.Reader
+}
+
+// NewBuildGradleVersionUpdater constructs a new BuildGradleVersionUpdater.
+func NewBuildGradleVersionUpdater(buildGradleReader io.Reader) BuildGradleVersionUpdater {
+	return BuildGradleVersionUpdater{buildGradleReader: buildGradleReader}
+}
+
+// UpdateResult stors the result of the version update.
+type UpdateResult struct {
+	NewContent          string
+	FinalVersionCode    string
+	FinalVersionName    string
+	UpdatedVersionCodes int
+	UpdatedVersionNames int
+}
+
+// UpdateVersion executes the version updates.
+func (u BuildGradleVersionUpdater) UpdateVersion(newVersionCode, versionCodeOffset int, newVersionName string) (UpdateResult, error) {
+	res := UpdateResult{}
+	var err error
+
+	res.NewContent, err = findAndUpdate(u.buildGradleReader, map[*regexp.Regexp]updateFn{
+		regexp.MustCompile(versionCodeRegexPattern): func(line string, lineNum int, match []string) string {
+			oldVersionCode := match[1]
+			res.FinalVersionCode = oldVersionCode
+			updatedLine := ""
+
+			if newVersionCode > 0 {
+				res.FinalVersionCode = strconv.Itoa(newVersionCode + versionCodeOffset)
+				updatedLine = strings.Replace(line, oldVersionCode, res.FinalVersionCode, -1)
+				res.UpdatedVersionCodes++
+				log.Printf("updating line (%d): %s -> %s", lineNum, line, updatedLine)
+			}
+
+			return updatedLine
+		},
+
+		regexp.MustCompile(versionNameRegexPattern): func(line string, lineNum int, match []string) string {
+			oldVersionName := match[1]
+			res.FinalVersionName = oldVersionName
+			updatedLine := ""
+
+			if newVersionName != "" {
+				quotedNewVersionName := newVersionName
+				if !(strings.HasPrefix(quotedNewVersionName, `"`) && strings.HasSuffix(quotedNewVersionName, `"`)) {
```

Comment:
what is the suffix is missing? `"1.2.3`

Reply:
In this case, the step will insert an invalid versionName (`""1.2.3"`) and most likely Gradle will fail on such build.gradle file.

Reply:
The intention was to remove trailing and leading double quotes, if any of them is missing, and add new leading and trailing `"`.
However in this commit: https://github.com/bitrise-steplib/steps-change-android-versioncode-and-versionname/pull/31/commits/acd4d23984c52ca304e1b4c7e50407a184bd4c7a I just removed the trailing unintentionally.

Fixed in the latest commit.